static/js/components/PodcastPlayButton.js

```diff
@@ -14,25 +19,60 @@ type Props = {
 export default function PodcastPlayButton(props: Props) {
   const { episode } = props
 
-  const dispatch = useDispatch()
+  const audioPlayerState = useSelector(audioPlayerStateSelector)
+  const currentlyPlayingAudio = useSelector(currentlyPlayingAudioSelector)
+
+  const episodeAudioInitialized = currentlyPlayingAudio.url === episode.url
+  const episodeCurrentlyPlaying =
+    episodeAudioInitialized && audioPlayerState === AUDIO_PLAYER_PLAYING
+
   const initAudioPlayer = useInitAudioPlayer({
     title:       episode.podcast_title,
     description: episode.title,
     url:         episode.url
   })
-  const playClick = useCallback(
+
+  const initializeAudioPlayer = useCallback(
     e => {
       e.stopPropagation()
       e.preventDefault()
       initAudioPlayer()
     },
-    [dispatch]
+    [initAudioPlayer]
   )
 
+  const togglePlayState = e => {
+    e.stopPropagation()
+    e.preventDefault()
+
+    const audioElement = Amplitude.getAudio()
+    if (audioPlayerState === AUDIO_PLAYER_PLAYING) {
+      audioElement.pause()
+    } else {
+      audioElement.play()
+    }
+  }
+
   return (
-    <div className="podcast-play-button black-surround" onClick={playClick}>
-      Play
-      <i className="material-icons play_arrow">play_arrow</i>
+    <div
+      className={`podcast-play-button ${
+        episodeCurrentlyPlaying ? "grey-surround" : "black-surround"
```

Comment:
Minor nit:

Can we clean this up a bit by defining a variable above the return statement with the result of the conditional and just include that here?  That way the opening div tag could maybe be on one line and this would read more like HTML.

Reply:
sure